storage/kv/txn.go

```diff
@@ -81,3 +81,19 @@ func (t badgerTxn) GetAll(prefix []byte, handler func(int, basedb.Obj) error) er
 func (t badgerTxn) Delete(prefix []byte, key []byte) error {
 	return t.txn.Delete(append(prefix, key...))
 }
+
+func (t badgerTxn) CountByCollection(prefix []byte) (int64, error) {
+	return t.db.CountByCollection(prefix)
```

Comment:
I think this should be a part of the transaction (same for methods below)

Reply:
Do you mean on *badger.Txn? but it doesnt implement these methods. Can you please clarify a bit?

Reply:
Yes, it doesn't, therefore @moshe-blox left them in the `basedb.IDb` interface (now `basedb.Database`). Can we remove them from `badgerTxn` and move these methods from transactions to `basedb.Database`?

Reply:
Moved back to Database interface